### ISSUE TITLE
Upsert mutation fails to link entry with category

### ISSUE THREAD

**Issue by kristian-wcd** [markhuot/craftql#116]
Hello! First of all, huge thank you for the plugin, works out great so far. However, when I try to create/update an entry and link it to a already defined category, I run into the following error:
`
{
  "error": "Undefined index: title"
}
`
**How to reproduce this error:**
1. Create a category group (e.g. Pet)
2. Create a category in the Pet group (e.g. Dog)
3. Create a new Section (e.g. Animal)
4. Setup entry type for that section (e.g. also named Animal)

The entry type animal has a firstname (Text) field and a petType, which is a reference to a pet category.

Having the setup from above, I get the error that I already showed above after running this query:

`
mutation {
	upsertAnimal (firstname: "Charlie", petType: {id: 401}) {
                id
		firstname
	}
}
`

Would be really glad if we could fix this. Also, are there any error logs from the plugin? Those would be helpful as well.

**Comment by markhuot** [comment]
If you update to the latest "nightly" version, `dev-master` you should see this fixed. The issue was in how I was treating existing v.s. new categories.

I'll leave this ticket open until this fix makes its way in to a stable tag.

**Comment by kristian-wcd** [comment]
Thanks, sounds great! Also, it's somewhat off topic, but is there (or will there be) a way to create/update categories?

**Comment by markhuot** [comment]
@kristian-wcd, you can actually create new categories while you set them. The following would create a new category as well as re-use an existing category.

```graphql
mutation {
  upsertEntryType(
    customCategoriesField:[{title:"New Category"}, {id:123}]
  ) {
    id
  }
}
```

**Comment by kristian-wcd** [comment]
@markhuot I see, thanks :) What about category groups?

**Comment by kristian-wcd** [comment]
@markhuot I just figured out that you indeed can create a new category, but it will not be automatically linked to the entry. In case of the code you provided it will create a new category "New Category", but link only the one with the id 123.

**Comment by markhuot** [comment]
Closing this issue. The fix is included in version 1.2.1.

**Comment by ashtonlance** [comment]
@kristian-wcd I know this issue is closed but have you successfully been able to use this?
With the code below, I get an `  "error": "A non well formed numeric value encountered"`

```
 mutation addProject($title: String!, $totalHours: Int!, $selectedCat: [GroupsCategoryInput]!)
    {
        upsertProjects(
            title: $title
            projectName: $title
            totalHours: $totalHours
            groups: $selectedCat
            
        ) {
        id
        title
        totalHours
    }
}

Variables:
{
  "title": "Test Member",
  "totalHours": 40,
  "selectedCat": [
    {
      "id": 48
    }
  ]
}
```
